subsys/net/lib/openthread/platform/radio.c

```diff
@@ -361,12 +443,49 @@ otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
 	return &sTransmitFrame;
 }
 
+static void get_rssi_energy_detected(struct device *dev, s16_t max_ed)
+{
+	ARG_UNUSED(dev);
+	radio_signal_set(RADIO_SIGNAL_BLOCKING, max_ed);
+}
+
 int8_t otPlatRadioGetRssi(otInstance *aInstance)
 {
-	ARG_UNUSED(aInstance);
+	s8_t ret_rssi = INT8_MAX;
+	int signal_value;
+
+	otRadioCaps radioCaps =  otPlatRadioGetCaps(aInstance);
+
+	if (!(radioCaps & OT_RADIO_CAPS_ENERGY_SCAN)) {
+		/*
+		 *TODO: No API in Zephyr to get the RSSI
+		 * when OT_RADIO_CAPS_ENERGY_SCAN is not available
+		 */
+		ret_rssi = 0;
+	} else {
+		/*
+		 * Blocking implementation of get RSSI
+		 * using no-blocking ed_scan
+		 */
+		int error = 0;
+		const u16_t energy_detection_time = 1;
+
+		error = radio_api->ed_scan(radio_dev, energy_detection_time,
+					    get_rssi_energy_detected);
+
+		if (!error) {
+
+			radio_signal_poll(RADIO_SIGNAL_BLOCKING, 1,
+							K_FOREVER);
```

Comment:
Allignment is off here and in many other places. If you break the argument list over a few lines, keep them aligned to the opening bracket.